nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -316,6 +322,12 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
                     'Expert parallelism is currently not supporting Apex distributed optimizer, use Mcore distributed optimizer instead'
                 )
 
+        if self.cfg.get('num_layers', 12) % self.cfg.get('pipeline_model_parallel_size', 1) != 0:
```

Comment:
It's a nitpick but I don't think we should set a default value for num_layers. I think it's safe to assume that that value is set in the config.